From 37b7a9cea10287f81b39e2c1e47cf57c0415cbbd Mon Sep 17 00:00:00 2001
From: Ankit Agrawal <ankit.agarwal@intel.com>
Date: Tue, 19 Dec 2023 09:52:39 +0530
Subject: [PATCH] Fixed for Crash in MD Launcher during Monkey Test.

Observed below crash
java.lang.NullPointerException Attempt to invoke virtual method
'java.lang.Object android.car.Car.getCarManager(java.lang.String)' on a
null object reference.

Removing unused code which was causing crash.

Tests:
Run monkey test and observe, test is pass.

Tracked-On: OAM-114266
Signed-off-by: Ankit Agrawal <ankit.agarwal@intel.com>
---
 .../multidisplay/launcher/LauncherActivity.java | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/tests/MultiDisplaySecondaryHomeTestLauncher/src/com/android/car/multidisplay/launcher/LauncherActivity.java b/tests/MultiDisplaySecondaryHomeTestLauncher/src/com/android/car/multidisplay/launcher/LauncherActivity.java
index a14e2594fd..15b80318b2 100644
--- a/tests/MultiDisplaySecondaryHomeTestLauncher/src/com/android/car/multidisplay/launcher/LauncherActivity.java
+++ b/tests/MultiDisplaySecondaryHomeTestLauncher/src/com/android/car/multidisplay/launcher/LauncherActivity.java
@@ -97,8 +97,6 @@ public class LauncherActivity extends FragmentActivity implements AppPickedCallb
     private boolean mAppDrawerShown;
 
     private InputManager mInputManager;
-    private CarOccupantZoneManager mZoneManager;
-    private Car mCar;
     private int mDisplayId;
     private LinearLayout mOverlay;
     private Button mBackKey;
@@ -113,9 +111,6 @@ public class LauncherActivity extends FragmentActivity implements AppPickedCallb
         mScrimView = findViewById(R.id.Scrim);
         mAppDrawerView = findViewById(R.id.FloatingSheet);
         mInputManager = (InputManager) getSystemService(Context.INPUT_SERVICE);
-        mCar = Car.createCar(getApplicationContext());
-	mZoneManager = (CarOccupantZoneManager) mCar.getCarManager(Car.CAR_OCCUPANT_ZONE_SERVICE);
-        mDisplayId = getLauncherDisplayId();
         if (android.os.Process.myUserHandle().getIdentifier() != ActivityManager.getCurrentUser()) {
             createNavigationKeysOverlay(mOverlay != null);
         }
@@ -194,18 +189,6 @@ public class LauncherActivity extends FragmentActivity implements AppPickedCallb
         });
     }
 
-    private int getLauncherDisplayId() {
-        List<CarOccupantZoneManager.OccupantZoneInfo> zonelist = mZoneManager.getAllOccupantZones();
-        for (CarOccupantZoneManager.OccupantZoneInfo zone : zonelist) {
-            int passengerUserId = android.os.Process.myUserHandle().getIdentifier();
-            if (passengerUserId == mZoneManager.getUserForOccupant(zone)) {
-                Display display = mZoneManager.getDisplayForOccupant(zone, CarOccupantZoneManager.DISPLAY_TYPE_MAIN);
-                if (display != null)
-                    return display.getDisplayId();
-            }
-        }
-        return Display.INVALID_DISPLAY;
-    }
 
     private void createNavigationKeysOverlay(boolean isCreated) {
         if (isCreated) return;
-- 
2.17.1

